src/Scenario/Resolver.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Scenario;
+
+use Magento\MagentoCloud\App\ContainerInterface;
+use Magento\MagentoCloud\Process\ProcessInterface;
+use Magento\MagentoCloud\Scenario\Exception\ValidationException;
+
+/**
+ * Resolve step arguments.
+ */
+class Resolver
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * @param ContainerInterface $container
+     */
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * @param array $scenarios
+     * @return ProcessInterface[]
+     * @throws ValidationException
+     */
+    public function resolve(array $scenarios): array
+    {
+        $instances = [];
+
+        foreach ($scenarios as $step) {
+            $instance = $this->container->create(
+                $step['type'],
+                $this->resolveParams($step['arguments'] ?? [])
+            );
+
+            if (!$instance instanceof ProcessInterface) {
+                throw new ValidationException(sprintf(
+                    '%s is not instance of %s',
+                    get_class($instance),
+                    ProcessInterface::class
+                ));
+            }
+
+            $instances[$step['name']] = $instance;
+        }
+
+        return $instances;
+    }
+
+    /**
+     * @param array $data
+     * @return array
+     * @throws ValidationException
+     */
+    private function resolveParams(array $data): array
+    {
+        $newData = [];
+
+        foreach ($data as $item) {
+            $type = $item['xsi:type'] ?? null;
+            $name = $item['name'] ?? null;
+
+            if (!$name) {
+                throw new ValidationException('Empty parameter name');
```

Comment:
It will be difficult to find destination of this parameter. At least add its name, and it possible name of scenario or step

Reply:
I can display a name when it's empty. Added step name